ports/stm32/boards/LEGO_HUB_NO6/spiflash.py

```diff
@@ -4,18 +4,34 @@
 from micropython import const
 
 _PAGE_SIZE = const(256)  # maximum bytes writable in one SPI transfer
+_CMD_WRITE = const(0x02)
+_CMD_READ = const(0x03)
 _CMD_RDSR = const(0x05)
 _CMD_WREN = const(0x06)
 _CMD_WRITE_32 = const(0x12)
 _CMD_READ_32 = const(0x13)
+_CMD_SEC_ERASE = const(0x20)
 _CMD_SEC_ERASE_32 = const(0x21)
-_CMD_C4READ_32 = const(0xEC)
+_CMD_JEDEC_ID = const(0x9F)
 
 
 class SPIFlash:
     def __init__(self, spi, cs):
         self.spi = spi
         self.cs = cs
+        self.id = self._get_id()
+        # flash chip on Hub No. 6 uses 32-bit addressing
+        _32_bit = self.id == b"\xef\x40\x19"
+        self._READ = _CMD_READ_32 if _32_bit else _CMD_READ
+        self._WRITE = _CMD_WRITE_32 if _32_bit else _CMD_WRITE
+        self._ERASE = _CMD_SEC_ERASE_32 if _32_bit else _CMD_SEC_ERASE
+
+    def _get_id(self):
```

Comment:
With this implemented, I think you can extract the SPI flash size from it.  That could be used instead of `pyb.Flash(start=0)`.  (This is only an observation, no need to change anything, unless you want to.)